backend/server.py

```diff
@@ -219,52 +241,51 @@ def add_status():
 
     global status_data
 
-    while status_data is None:
-        pass
-
     return_data = status_data
     status_data = {}
     status_data['nodes'] = []
 
-
     return jsonify(return_data)
 
 @app.route('/fetcher/statistics', methods=['GET', 'POST'])
 def add_statistics():
-
     global statistics_data
 
-    while statistics_data is None:
-        pass
+    reason=waiter_statistics.wait(default_values.TIMEOUT_STATISTICS)
+    if reason==AwakeReason.timeout:
+        print('Timeout reached. This error may be due to the fact that the inference node is not available or no new statistics have been received.')

```

Comment:
The error message is a bit redundant here. Mentioning both the unavailability of the inference node and the lack of new statistics isn't necessary, as the absence of statistics already implies that no new node is available. Additionally, it’s the sender node that is responsible, not the inference node.